govcd/any_edgegateway.go

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// AnyEdgeGateway is a common structure which fetches any type of Edge Gateway (NSX-T or NSX-V)
+// using OpenAPI endpoint. It can be useful to identify type of Edge Gateway or just retrieve common
+// fields - like OwnerRef. There is also a function GetNsxtEdgeGateway to convert it to
+// NsxtEdgeGateway (if it is an NSX-T one)
+type AnyEdgeGateway struct {
+	EdgeGateway *types.OpenAPIEdgeGateway
+	client      *Client
+}
+
+// GetNsxtEdgeGatewayById allows retrieving NSX-T edge gateway by ID for Org admins
```

Comment:
I may have understood this wrong, but isn't this comment referring to another function and not `GetAnyEdgeGatewayById `? As I understood, it "retrieves either an NSX-T or NSX-V gatewat by ID".

Reply:
Good spot - improved it